pkg/storegateway/indexheader/index/postings.go

```diff
@@ -257,6 +272,23 @@ func (t *PostingOffsetTableV1) LabelValues(name string, prefix string, filter fu
 	return values, nil
 }
 
+func (t *PostingOffsetTableV1) LabelValuesOffsets(name, prefix string, filter func(string) bool) ([]PostingListOffset, error) {
+	e, ok := t.postings[name]
+	if !ok {
+		return nil, nil
+	}
+	values := make([]PostingListOffset, 0, len(e))
+	for k, r := range e {
+		if strings.HasPrefix(k, prefix) && (filter == nil || filter(k)) {
+			values = append(values, PostingListOffset{Value: k, Off: r})
+		}
+	}
+	sort.Slice(values, func(i, j int) bool {
+		return values[i].Value < values[j].Value
+	})
```

Comment:
Do we really need these to be sorted? This is an expensive contract. 
If we do, lets state it in the docs too.

Reply:
I think we need them sorted for the LabelValues RPC - here https://github.com/grafana/mimir/blob/71b5cf53a059744d34014b5a455b9ebb3fd6a6a2/pkg/storegateway/bucket.go#L1139-L1170 expects that the stirng values are sorted and ther code here doesn't sort them https://github.com/grafana/mimir/blob/71b5cf53a059744d34014b5a455b9ebb3fd6a6a2/pkg/storegateway/bucket.go#L1219-L1231

I think we should sort them only for the LabelValues use-case, not on Series calls as well. I can maybe open another PR, even though that's only for V1 headers, which should be less common

Reply:
But this is `LabelValuesOffsets` which isn't used for `LabelValues` RPC right?
I agree that it's less important for V1.
Also, one argument could be that V2 provides them sorted, so we should match in V1.
Okay 👍

Reply:
i'll keep the same behaviour and document it in this PR